target/linux/lantiq/dts/RS353.dts

```diff
@@ -0,0 +1,318 @@
+/dts-v1/;
+
+/include/ "vr9.dtsi"
+
+/ {
+	model = "RS353 - Bintec Elmeg RS353";
+
+	chosen {
+		bootargs = "console=ttyLTQ0,115200 init=/etc/preinit";
+		leds {
+                        running = &status;
+                        boot = &status;
+			wifi = &wlan;
+			usb2 = &usb; /* Front port is mapped to device 2 */
+			dsl = &dsl;
+                };
+
+	};
+
+	memory@0 {
+		reg = <0x0 0x8000000>;
+	};
+
+	sram@1F000000 {
+		cgu0: cgu@103000 {
+                        compatible = "lantiq,cgu-xway";
+                        reg = <0x103000 0x1000>;
+                        #clock-cells = <1>;
+			lantiq,phy-clk-src = <0x2>;
+		};
+	};
+
+	fpi@10000000 {
+		localbus@0 {
+			nor-boot@0 {
+				compatible = "lantiq,nor";
+				bank-width = <2>;
+				reg = <0 0x0 0x2000000>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				partitions {
+					compatible = "fixed-partitions";
+					#address-cells = <1>;
+					#size-cells = <1>;
+
+					boardconfig: partition@0 {
```

Comment:
> In my opinion, the kernel relocation is the better option. I used U-boot for simplicity and to be sure I can launch it from the original Bootmonitor.

I would prefer to get rid of the extra u-boot and use already existing OpenWrt components. I like your approach that you are changing only the minimum requires bits and it's easily possible to go back to the bintec firmware.

> The image length in the header determines the length for crc32 calculation (as you said), but it also determines the length of image to be really writed to the flash. So, I can't use it to bypass the crc32 calculation by this way. Also, I can patch and reflash the bootmonitor to boot even if the crc32 is not correct.

Okay, the problem of the image length parameter is only relevant if you use the tftp(?) install or flash OpenWrt via the bintec webinterface, correct? Afterwards the OpenWrt sysupgrade code is used which doesn't validate the boss header.

You must have had the same problems with the installation of uboot. How have you done the first install of uboot + OpenWrt? Do you have an idea how to workaround the issue for the first installation of OpenWrt?

Do you plan to give the kernel relocation + kernel with gzip a try?